packages/web/src/pages/upload-page/fields/StemsAndDownloadsMenuFields.tsx

```diff
@@ -227,6 +245,29 @@ export const StemsAndDownloadsMenuFields = () => {
     }
   }, [downloadRequiresFollow, setIsDownloadable])
 
+  // Allow full track download and provide lossless files if additional are uploaded for the first time.
+  useEffect(() => {
+    if (
+      firstTimeStemsUploaded &&
+      stemsValue.length > 0 &&
+      previousStemsValue &&
+      previousStemsValue.length < stemsValue.length
```

Comment:
nit: do we need to check `previousStemsValue` here? is `firstTimeStemsUploaded && stemsValue.length > 0` not sufficient?

Reply:
no, because what if we are editing a track that already had stems?